addrmgr/addrmanager.go

```diff
@@ -954,7 +947,6 @@ func (a *AddrManager) Good(addr *wire.NetAddress) {
 
 	// replace with ka in list.
 	ka.tried = true
-	entry.Value = ka
```

Comment:
The intent here was to replace the evicted entry, but that is no longer happening.  I believe what you'll need to do is change `pickTicket` to return which index to evict instead of  `*KnownAddress`.  Then, above you can do:

```Go
evictIdx ;= a.pickTried(bucket)
rmka := a.addrTried[bucket][evictIdx]
...
ka.tried = true
a.addrTried[bucket][evictIdx] = ka
```